doc/sfputil/read_write_eeprom_by_page.md

```diff
@@ -0,0 +1,191 @@
+
+Example:
+```
+sfputil read-eeprom Ethernet0 0 255 1               # valid
+sfputil read-eeprom Ethernet0 0 255 2               # invalid size, out of range, 255+2=257 is not a valid offset
+sfputil read-eeprom Ethernet0 0 256 1               # invalid offset 256 for page 0, must be in range [0, 255]
+sfputil read-eeprom Ethernet0 1 0 1                 # invalid offset 0 for page 1, must be >=128
+```
+
+##### sff8436 and sff8636 validation
+
+Passive cable:
+- Valid page: 0
+- Valid offset: 0-255
+
+Active cable:
+- Valid page: 0-255
+- Valid offset: page 0 (0-255), other (128-255)
+
+For active cable, there is no "perfect" page validation as it is too complicated. User is responsible to make sure the page existence according to cable user manual.
+
+```
+sfputil write-eeprom Ethernet0 0 255 ff            # valid
+sfputil write-eeprom Ethernet0 0 255 ff00          # invalid size, out of range, 255+2=257 is not a valid offset
+sfputil write-eeprom Ethernet0 0 256 ff            # invalid offset 256 for page 0, must be in range [0, 255]
+sfputil write-eeprom Ethernet0 1 0 ff              # invalid offset 0 for page 1, must be >=128
+```
+
+##### sff8472 validation
+
+Passive cable:
+- Valid wire address: [A0h] (case insensitive)
+- Valid offset: A0h (0-128)
+
+Active cable:
+- Valid wire address: [A0h, A2h] (case insensitive)
+- Valid offset: A0h (0-255), A2h (0-255)
+
+```
+sfputil read-eeprom Ethernet0 0 0 1 --wire-addr a0h               # valid
+sfputil read-eeprom Ethernet0 0 0 2 --wire-addr A0h               # invalid size, out of range, 255+2=257 is not a valid offset
+sfputil read-eeprom Ethernet0 1 0 1 --wire-addr a2h               # invalid offset 256 for page 0, must be in range [0, 255]
+sfputil read-eeprom Ethernet0 1 0 1 --wire-addr a0h               # invalid offset 0 for page 1, must be >=128
+```
+
+### SAI API
+
+N/A
+
+### Configuration and management
+
+#### Manifest (if the feature is an Application Extension)
+
+N/A
+
+#### CLI/YANG model Enhancements
+
+##### sfputil read-eeprom
+
+```
+admin@sonic:~$ sfputil read-eeprom --help
+Usage: sfputil read-eeprom [OPTIONS] <port_name> <page> <offset> <size>
+
+  Read SFP EEPROM data
+
+Options:
+  --no-format             Display non formatted data.
+  --wire-addr             Wire address of sff8472.
+  --help                  Show this message and exit.
+```
+
+Example:
+
+```
+sfputil read-eeprom Ethernet0 0 100 2
+        00000064 4a 44                                            |..|
+
+sfputil read-eeprom Ethernet0 0 0 32
+        00000000 11 08 06 00 00 00 00 00  00 00 00 00 00 00 00 00 |................|
+        00000010 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 |................|
+
+sfputil read-eeprom Ethernet0 0 100 2 --no-format
+4a44
+```
+
+##### sfputil write-eeprom
+
+```
+admin@sonic:~$ sfputil write-eeprom --help
+Usage: sfputil write-eeprom [OPTIONS] <port> <page> <offset> <data>
+
+  Write SFP EEPROM data
+

```

Comment:
@Junchao-Mellanox please specify in the help option what does the <port> mean. most user may consider it as logical port name

Reply:
click.argument does not support help message. Check https://github.com/pallets/click/issues/587. How about following:

```
Usage: sfputil read-eeprom [OPTIONS] <logical_port_name> <page> <offset> <size>

  Read SFP EEPROM data

Options:
  --no-format       Display non formatted data
  --wire-addr TEXT  Wire address of sff8472
  --help            Show this message and exit.
```